ext/app/electron/config.ts

```diff
@@ -0,0 +1,241 @@
+          log.warn(`${confKey} has invalid value ${confValue}, using default value ${defaultValue}`);
+          process.env[envKey] = defaultValue;
+        } else {
+          process.env[envKey] = confValue.toString();
+        }
+      }
+    } else {
+      // envvar is defined, ignore config
+      if (confValue !== undefined) {
+        log.warn(`${confKey} is overridden by $${envKey}`);
+      }
+      if (!validator(envValue)) {
+        log.warn(`$${envKey} has invalid value ${envValue}, using default value ${defaultValue}`);
+        process.env[envKey] = defaultValue;
+      }
+    }
+  }
+}
+
+
+export async function loadConfig(filename: string = DEFAULT_CONFIG_FILE) {
+  if (process.env.GRIST_ELECTRON_AUTH !== undefined) {
+    log.warn("GRIST_ELECTRON_AUTH has been deprecated; use GRIST_DESKTOP_AUTH instead.");
+  }
+  let config: Config;
+  try {
+    const configBuffer = fse.readFileSync(filename);
+    config = new Config(ini.parse(configBuffer.toString("utf8")));
+  } catch (err) {
+    if (err instanceof Error && err.message.startsWith("ENOENT")) {
+      log.warn(`Configuration file ${filename} does not exist, using default config.`);
+      config = new Config({});
+    } else {
+      log.error(`Failed to read configuration file: ${err}`);
+      process.exit(1);
+    }
+  }
+  // Section: login
+  config.apply(
+    "login.username",
+    "GRIST_DEFAULT_USERNAME",
+    NO_VALIDATION,
+    getUsername()
+  );
+  config.apply(
+    "login.email",
+    "GRIST_DEFAULT_EMAIL",
+    NO_VALIDATION,
+    getEmail()
+  );
+  // Section: server
+  config.apply(
+    "server.listen",
+    "GRIST_HOST",
+    NO_VALIDATION,
+    "localhost"
+  );
+  config.apply(
+    "server.port",
+    "GRIST_PORT",
+    (portstr) => {
+      if (! /^\d+$/.test(portstr)) {
+        return false;
+      }
+      const port = parseInt(portstr);
+      return port > 0 && port < 65536;
+    },
+    (await getAvailablePort(47478)).toString()
+  );
+  config.apply(
+    "server.auth",
+    "GRIST_DESKTOP_AUTH",
+    (auth) => ["strict", "none", "mixed"].includes(auth),
+    "strict"
+  );
+  // Section: sandbox
+  config.apply(
+    "sandbox.flavor",
+    "GRIST_SANDBOX_FLAVOR",
+    (flavor) => ["pyodide", "unsandboxed", "gvisor", "macSandboxExec"].includes(flavor),
+    "pyodide"
+  );
+  // Section: storage
+  config.apply(
+    "storage.instance",
+    "GRIST_INST_DIR",
+    NO_VALIDATION,
+    APPDATA_DIR
+  );
+  config.apply(
+    "storage.documents",
+    "GRIST_DATA_DIR",
+    NO_VALIDATION,
+    electron.app.getPath("documents")
+  );
+  config.apply(
+    "storage.plugins",
+    "GRIST_USER_ROOT",
+    NO_VALIDATION,
+    path.join(electron.app.getPath("home"), ".grist")
+  );
+  config.apply(
+    "storage.homedb",
+    "TYPEORM_DATABASE",
+    NO_VALIDATION,
+    path.join(APPDATA_DIR, "home.sqlite3")
+  );
+
+  const homeDBLocation = path.parse((process.env.TYPEORM_DATABASE as string)).dir;
```

Comment:
Where did we land on trying to preserve/copy the existing home db? From my perspective as a know-nothing user, I just upgraded my version of Grist and "lost all my files".

I see that even if I find and copy across my old home db, I have still "lost all my files" because I'm a new and different user.

If I learn about the config file and set `login.email` to `you@example.com` I have still "lost all my files" because my login as the new and different user is remembered.

If I log out and log back in again, I can finally see my files.

A little bit worried about a wave of support requests as people upgrade, none of the above steps would be very obvious.

We could grit our teeth and just go through with it, is that your thinking? I know you have a modal planned.